stdlib/3/tracemalloc.pyi

```diff
@@ -64,21 +39,30 @@ class StatisticDiff:
     traceback: Traceback
     def __init__(self, traceback: Traceback, size: int, size_diff: int, count: int, count_diff: int) -> None: ...
 
+_FrameTupleT = Tuple[str, int]
+
+class Frame:
+    filename: str
+    lineno: int
+    def __init__(self, frame: _FrameTupleT) -> None: ...
+
+if sys.version_info >= (3, 9):
+    _TraceTupleT = Union[Tuple[int, int, Sequence[_FrameTupleT], Optional[int]], Tuple[int, int, Sequence[_FrameTupleT]]]
```

Comment:
What is the Union for? On master it seems to always return a four-tuple from `trace_to_pyobject`.

Reply:
I typed this based on the comments and the usage within tracemalloc.py.
This comment might just be out of date, but it isn't reassuring: https://github.com/python/cpython/blob/eaca2aa117d663acf8160a0b4543ee2c7006fcc7/Lib/tracemalloc.py#L279
I feel like if it was always a four-tuple, it would be better to explicitly pass the arguments around, rather than slicing like this: https://github.com/python/cpython/blob/eaca2aa117d663acf8160a0b4543ee2c7006fcc7/Lib/tracemalloc.py#L293
Plus, if someone is make their own `Trace` objects out there, as far as I can tell, three-tuples should still work.

Reply:
hm, perhaps strictly speaking it should have different types for the return and argument positions. Although it feels like we're going too deep into a rabbit hole here here :)